### ISSUE TITLE
  Error: Expected package major version to equal @build_bazel_rules_nodejs major version

### ISSUE THREAD

**Issue by megamegabits** [Dig-Doug/rules_typescript_proto#132]
## Description
Seeing a similar error to [issue-695365869](https://github.com/Dig-Doug/rules_typescript_proto/issues/129#issue-695365869). Not sure if rules_nodejs needs another version bump, or if the error is on my side. Thank you! 

## Errors seen
```
8:21:52 ❯ bazel build //src/protos:empty_ts_proto
INFO: Repository rules_typescript_proto_deps instantiated at:
  no stack (--record_rule_instantiation_callstack not enabled)
Repository rule yarn_install defined at:
  /home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl:424:31: in <toplevel>
ERROR: An error occurred during the fetch of repository 'rules_typescript_proto_deps':
   Traceback (most recent call last):
	File "/home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 417, column 13, in _yarn_install_impl
		fail("yarn_install failed: %s (%s)" % (result.stdout, result.stderr))
Error in fail: yarn_install failed: yarn install v1.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.1.2: The platform "linux" is incompatible with this module.
info "fsevents@2.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
 (warning " > @bazel/karma@1.7.0" has unmet peer dependency "jasmine-core@>=2.0.0".
warning " > @bazel/rollup@1.7.0" has incorrect peer dependency "rollup@>=1.0.0 <2.0.0".
error /home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma: Command failed.
Exit code: 1
Command: node npm_version_check.js
Arguments: 
Directory: /home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma
Output:
/home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma/npm_version_check.js:17
  throw new Error(`Expected package major version to equal @build_bazel_rules_nodejs major version
  ^

Error: Expected package major version to equal @build_bazel_rules_nodejs major version
    @bazel/karma - 1.7.0  
    @build_bazel_rules_nodejs - 2.2.2
  See https://github.com/bazelbuild/rules_nodejs/wiki/Avoiding-version-skew
    at Object.<anonymous> (/home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma/npm_version_check.js:17:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
)
ERROR: /home/meganvw/src/meganvw/bazel-example-client/src/protos/BUILD:11:25: //src/protos:empty_ts_proto depends on @rules_typescript_proto_deps//ts-protoc-gen/bin:protoc-gen-ts in repository @rules_typescript_proto_deps which failed to fetch. no such package '@rules_typescript_proto_deps//ts-protoc-gen/bin': yarn_install failed: yarn install v1.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.1.2: The platform "linux" is incompatible with this module.
info "fsevents@2.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
 (warning " > @bazel/karma@1.7.0" has unmet peer dependency "jasmine-core@>=2.0.0".
warning " > @bazel/rollup@1.7.0" has incorrect peer dependency "rollup@>=1.0.0 <2.0.0".
error /home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma: Command failed.
Exit code: 1
Command: node npm_version_check.js
Arguments: 
Directory: /home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma
Output:
/home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma/npm_version_check.js:17
  throw new Error(`Expected package major version to equal @build_bazel_rules_nodejs major version
  ^

Error: Expected package major version to equal @build_bazel_rules_nodejs major version
    @bazel/karma - 1.7.0  
    @build_bazel_rules_nodejs - 2.2.2
  See https://github.com/bazelbuild/rules_nodejs/wiki/Avoiding-version-skew
    at Object.<anonymous> (/home/meganvw/.cache/bazel/_bazel_meganvw/f297e2800e4b37f57e5a9008a9e00bc8/external/rules_typescript_proto_deps/node_modules/@bazel/karma/npm_version_check.js:17:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11
)
ERROR: Analysis of target '//src/protos:empty_ts_proto' failed; build aborted: Analysis failed
INFO: Elapsed time: 2.644s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
```

## Minimal Reproduction
```
load("@rules_proto//proto:defs.bzl", "proto_library")
load("@rules_typescript_proto//:index.bzl", "typescript_proto_library")

proto_library(
  name = "test_proto",
  srcs = [
    "test.proto",
  ],
)

typescript_proto_library(
  name = "test_ts_proto",
  proto = ":test_proto",
)
```

#### What operating system are you using?
Ubuntu 20.04 LTS

#### What version of bazel are you using?
3.5.0

#### What version of the library are you using?
51c7c5995f5de89ea1bbd64d956fd589f1c03357ab6768032930fadc2570f6a8

## Any other comments?
Thanks!

**Comment by Dig-Doug** [comment]
Hi,

Are you using the latest version of this library? I don't see commit `51c7c5995f5de89ea1bbd64d956fd589f1c03357ab6768032930fadc2570f6a8` in the history.

Maybe you are using version `0.0.5`. I hadn't cut a new release recently, I've just created a new one.

Can you try version `0.0.6` and see if you still see the problem?

**Comment by megamegabits** [comment]
Thanks for the reply! I updated to 0.0.6 and now see:
```
13:45:14 ❯ bazel build //src/protos:pizza_ts_proto
ERROR: /home/meganvw/src/meganvw/bazel-example-client/src/protos/BUILD:4:14: every rule of type proto_library implicitly depends upon the target '@rules_typescript_proto//src:change_import_style', but this target could not be found because of: error loading package '@rules_typescript_proto//src': Label '@build_bazel_rules_nodejs//packages/typescript:index.bzl' is invalid because 'packages/typescript' is not a package; perhaps you meant to put the colon here: '@build_bazel_rules_nodejs//:packages/typescript/index.bzl'?
ERROR: Analysis of target '//src/protos:pizza_ts_proto' failed; build aborted: Loading failed
INFO: Elapsed time: 0.264s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (2 packages loaded, 2 targets configured)
```

for reference this is my WORKSPACE file:
```
13:45:16 ▲❯ cat WORKSPACE
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: WORKSPACE
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ workspace(
   2   │     name = "bazel_example_client",
   3   │     managed_directories = {"@npm": ["src/node_modules"]},
   4   │ )
   5   │ 
   6   │ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
   7   │ 
   8   │ http_archive(
   9   │     name = "build_bazel_rules_nodejs",
  10   │     sha256 = "452bef42c4b2fbe0f509a2699ffeb3ae2c914087736b16314dbd356f3641d7e5",
  11   │     urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/2.3.0/rules_nodejs-2.3.0.tar.gz"],
  12   │ )
  13   │ load("@build_bazel_rules_nodejs//:index.bzl", "npm_install")
  14   │ npm_install(
  15   │     name = "npm", # Name this npm so that Bazel Label references look like @npm//package
  16   │     package_json = "//src:package.json",
  17   │     package_lock_json = "//src:package-lock.json",
  18   │ )
  19   │ 
  20   │ load("@build_bazel_rules_nodejs//:package.bzl", "rules_nodejs_dev_dependencies")
  21   │ rules_nodejs_dev_dependencies()
  22   │ 
  23   │ 
  24   │ http_archive(
  25   │     name = "rules_proto",
  26   │     sha256 = "602e7161d9195e50246177e7c55b2f39950a9cf7366f74ed5f22fd45750cd208",
  27   │     strip_prefix = "rules_proto-97d8af4dc474595af3900dd85cb3a29ad28cc313",
  28   │     urls = [
  29   │         "https://mirror.bazel.build/github.com/bazelbuild/rules_proto/archive/97d8af4dc474595af3900dd85cb3a29ad28cc313.tar.gz",
  30   │         "https://github.com/bazelbuild/rules_proto/archive/97d8af4dc474595af3900dd85cb3a29ad28cc313.tar.gz",
  31   │     ],
  32   │ )
  33   │ load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")
  34   │ rules_proto_dependencies()
  35   │ rules_proto_toolchains()
  36   │ 
  37   │ 
  38   │ http_archive(
  39   │     name = "rules_typescript_proto",
  40   │     sha256 = "cca55a4a8eb8489e8c8341355d9e69937685d831792d71c3051170e1c040d310",
  41   │     strip_prefix = "rules_typescript_proto-0.0.6",
  42   │     urls = [
  43   │         "https://github.com/Dig-Doug/rules_typescript_proto/archive/0.0.6.tar.gz",
  44   │     ],
  45   │ )
  46   │ load("@rules_typescript_proto//:index.bzl", "rules_typescript_proto_dependencies")
  47   │ rules_typescript_proto_dependencies()

```

**Comment by Dig-Doug** [comment]
I think this is happening because we switched to loading rules_nodejs from source.

Can you try loading rules_nodejs from source? There's an example [here](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/from_source).

**Comment by mc0** [comment]
> I think this is happening because we switched to loading rules_nodejs from source.
> 
> Can you try loading rules_nodejs from source? There's an example [here](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/from_source).

@Dig-Doug Thanks for the work on these rules.

I ran into this same problem and tried doing the "from source" dependency. That did indeed fix the issue with 0.0.6.

The "from source" method seems like it creates a fundamentally different state as compared to the packaged versions of rules_nodejs. It looks like the whole necessity of using the from source strategy is gone now that rules_nodejs has public visibility of rollup. Is that correct?

**Comment by Dig-Doug** [comment]
I haven't tried switching it back. It would be great to support both, but I would accept a PR switching back from using rules_nodejs from source.

Closing since this issue has been resolved.
